### PR TITLE
Core examples for simple small examples to files promise type docs

### PR DESCRIPTION
**Pull request by nickanderson**
Intended to replace the inline examples introduced to docs here: https://github.com/cfengine/documentation/pull/2877, see comment: https://github.com/cfengine/documentation/pull/2877#pullrequestreview-1101189833

### CODE REVIEW

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/8583/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/8583/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-8583/

**Review by larsewi — APPROVED** [review]
Typo in commit message, otherwise LGTM.

File: examples/files-content-with.cf [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+#@ test -e /tmp/hello && rm /tmp/hello
+#@ ```
+#+end_src
+#+begin_src cfengine3
+bundle agent __main__
+{
+  files:
+      "/tmp/hello"
+        content => "Output from stat: $(with)",
+        with => execresult( 'stat -c "%U" /', "useshell" );
```
Comment (olehermanse) [inline examples/files-content-with.cf:12]: I want to use a very simple shell command here:

```suggestion
        content => "Output from whoami: $(with)",
        with => execresult( "whoami", "useshell" );
```

**Comment by olehermanse** [comment]
@cf-bottom jenkins, please

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/8593/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/8593/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-8593/

**Comment by olehermanse** [comment]
@cf-bottom jenkins again, please

File: examples/files-content-with.cf [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  files:
+      "/tmp/hello"
+        content => "Output from whoami: $(with)",
+        with => execresult( "whoami", "useshell" );
+}
+#+end_src
+#+begin_src example_output
+#@ ```
+#@    info: Created file '/tmp/hello', mode 0600
+#@    info: Updated file '/tmp/hello' with content 'Output from whoami: root'
```
Comment (olehermanse) [inline examples/files-content-with.cf:18]:
```suggestion
        content => "Output from echo: $(with)",
        with => execresult( "echo hello", "useshell" );
}
#+end_src
#+begin_src example_output
#@ ```
#@    info: Created file '/tmp/hello', mode 0600
#@    info: Updated file '/tmp/hello' with content 'Output from echo: hello'
```

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/8595/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/8595/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-8595/

**Review by larsewi — APPROVED** [review]
ACK

**Comment by olehermanse** [comment]
@cf-bottom please test this again in Jenkins

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/8627/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/8627/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-8627/

**Comment by nickanderson** [comment]
@cf-bottom please test this again in Jenkins

**Comment by nickanderson** [comment]
Rebased.

**Comment by nickanderson** [comment]
@cf-bottom jenkins, please.

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/9444/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/9444/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-9444/

**Comment by nickanderson** [comment]
Alright, fixed up the examples to include inform true so that info level messages would be emitted and fixed up expected outputs.

**Comment by olehermanse** [comment]
@cf-bottom jenkins, please

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/10431/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/10431/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-10431/

**Comment by larsewi** [comment]
@cf-bottom jenkins please :)

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/10503/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/10503/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-10503/

**Comment by larsewi** [comment]
This line
```
info: Created file "/tmp/not-ubuntu_16-or-ubuntu_14", mode 0600
```
from `examples/files-create-class-expression.cf` is not printed on Ubuntu 16, and that's probably why it fails on Ubuntu 16

**Comment by nickanderson** [comment]
Alright, let's target linux instead ...

@cf-bottom jenkins please :)

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/10527/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/10527/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-10527/

**Comment by nickanderson** [comment]
@cf-bottom jenkins, again  please

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/10528/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/10528/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-10528/

**Comment by nickanderson** [comment]
@cf-bottom jenkins, again please

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/10530/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/10530/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-10530/

**Review by larsewi — APPROVED** [review]
Test failures are unrelated. Please squash commits ✅

**Pull request merged**